docs/source/guides/data.rst

```diff
@@ -0,0 +1,548 @@
+
+.. testsetup:: *
+
+    from pytorch_lightning.core.Lightning import LightningModule
+    from torch.utils.data import IterableDataSet
+    from pytorch_lightning.trainer.trainer.Trainer import Trainer
+
+.. _data:
+
+#############
+Managing Data
+#############
+
+Continue reading to learn about:
+
+* `Data Objects in Lightning <Data Objects in Lightning_>`_
+
+* `Models Defined by Data <Models Defined by Data_>`_
+
+* `Multiple Datasets <Multiple DataSets_>`_
+
+* `Sequential Data <Sequential Data_>`_
+
+*************************
+Data Objects in Lightning
+*************************
+
+There are a few different Data objects used in Lightning:
+
+.. list-table:: Data objects
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Object
+     - Definition
+   * - :class:`~torch.utils.data.Dataset`
```

Comment:
If you mention this one, You should also have a one-liner for `torch.utils.data.IterableDataset`